File: snippets/python/progress_bar.py
import time
import sys

def progress_bar(total=20, delay=0.1):
    for i in range(total + 1):
        bar = "#" * i + "-" * (total - i)
        sys.stdout.write(f"\r[{bar}] {int((i/total)*100)}%")
        sys.stdout.flush()
        time.sleep(delay)
    sys.stdout.write("\r\x1b[K✅ Done!\n")

progress_bar(30, 0.05)

File: snippets/nodejs/progress_bar.js
// Progress Bar in Terminal with Nodejs
function progressBar(total = 20, delay = 100) {
  return new Promise((resolve) => {
    let i = 0;
    const interval = setInterval(() => {
      i++;
      const filled = "#".repeat(i);
      const empty = "-".repeat(total - i);
      process.stdout.write(`\r[${filled}${empty}] ${Math.round((i / total) * 100)}%`);
      if (i === total) {
        clearInterval(interval);
        process.stdout.write("\r\x1b[K✅ Done!\n");
        resolve();
      }
    }, delay);
  });
}

// Example
(async () => {
  await progressBar(30, 50); // total 30 step, delay 50ms
})();
